cli/crates/cli/tests/apq.rs

```diff
@@ -0,0 +1,143 @@
+#![allow(unused_crate_dependencies, clippy::panic)]
+mod utils;
+
+use backend::project::GraphType;
+use serde_json::Value;
+use utils::environment::Environment;
+
+const SCHEMA: &str = r#"
+type Todo {
+    id: ID!
+    title: String
+}
+
+extend type Query {
+    todo(id: ID!): Todo @resolver(name: "todo")
+}
+"#;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn automatically_persisted_queries() {
+    let mut env = Environment::init_async().await;
+    env.grafbase_init(GraphType::Single);
+    env.write_schema(SCHEMA);
+    env.grafbase_dev_watch();
+    env.write_file(
+        "resolvers/todo.js",
+        r#"
+        export default function Resolver(_, {id}) {
+            if (id === "1") {
+                return {id: "1", title: "title"};
+            } else {
+                return null;
+            }
+        }
+        "#,
+    );
+
+    let client = env.create_async_client().with_api_key();
+
+    client.poll_endpoint(30, 300).await;
+
+    let query = r#"
+        query {
+            todo(id: "1") {
+                id
+                title
+            }
+        }
+    "#;
+
+    let execute = |query: &'static str, extensions: &Value| {
+        let request = client.gql::<Value>(query).extensions(extensions).into_reqwest_builder();
+        async move { request.send().await.unwrap().json::<Value>().await.unwrap() }
```

Comment:
There a reason this can't just be `client.gql::<Value>(query).extension(extensions).await.unwrap()`?

Reply:
assert_json_snapshot expect a `impl Serialize` which the `reqwest::Response` isn't